### ISSUE TITLE
Fuzzy searching is inconsistent between various pickers

### ISSUE THREAD

**Issue by lcrownover** [microsoft/vscode#194448]
I guess technically this could fall under a bug report or a feature request... please feel free to classify it however you see fit.

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.82.2
- OS Version: macOS 13.5.2

Steps to Reproduce:

When using the quick file open `cmd+p`, I correctly get fuzzy matching:

<img width="718" alt="image" src="https://github.com/microsoft/vscode/assets/15133041/bdcfca4d-a3fa-4828-ad8f-d3d9c10bf400">

However, when using the open recent picker, I don't get fuzzy matching:

<img width="718" alt="image" src="https://github.com/microsoft/vscode/assets/15133041/a7b73607-b77d-4b55-8013-803df62d83fa">

<img width="718" alt="image" src="https://github.com/microsoft/vscode/assets/15133041/8a1d70f4-a464-4ca1-98e0-0feff6351cff">

This is the same behavior with the new experimental.quickTextSearch (which is super awesome by the way!!).

<img width="718" alt="image" src="https://github.com/microsoft/vscode/assets/15133041/ac51a263-8d8d-4b93-b6dc-63195b4c6672">

<img width="718" alt="image" src="https://github.com/microsoft/vscode/assets/15133041/14df2340-2e9c-41ce-a321-f64cce3e0cc7">

There's another open issue that's similar to this, but I don't know if they're related: https://github.com/microsoft/vscode/issues/184679

**Comment by TylerLeonhardt** [comment]
I agree with you that it would be nice if the Open Recent aligned with `cmd+p` and I think we should track that feature request here.

However, the quickTextSearch doing that might be more contentious as that would break its alignment with the Search view experience. In any case, we should discuss that separately since that's a separate feature. Can you open a separate issue for quickTextSearch?
